prow/apis/prowjobs/v1/types.go

```diff
@@ -75,7 +75,9 @@ const (
 	// KnativeBuildAgent means prow will schedule the job via a build-crd resource.
 	KnativeBuildAgent ProwJobAgent = "knative-build"
 	// TektonAgent means prow will schedule the job via a tekton PipelineRun CRD resource.
-	TektonAgent = "tekton-pipeline"
+	TektonAgent ProwJobAgent = "tekton-pipeline"
+	// JenkinsXAgent means prow will schedule the jobs via JenkinsX pipeline.
+	JenkinsXAgent ProwJobAgent = "jenkins-x"
```

Comment:
So how does this actually end up working? Some proprietary controller that looks at ProwJobs in the cluster?

Reply:
That's more or less the size of it, but it won't be proprietary! :) But yeah that's more or less the size of it. Basically we'll be using a freestanding version (i.e. outside the prow tree, in its own repo) of the pipeline controller that Cosmin built at `prow/cmd/pipeline`, which will allow us to encapsulate the customizations we need without interfering with what you folks are doing upstream.

Reply:
cf. https://github.com/jenkins-x/jx/issues/4452